gridstatus/ercot.py

```diff
@@ -3351,3 +3353,81 @@ def _weather_zone_column_name_order(self):
+    def get_indicative_lmp_by_settlement_point(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        if date == "latest":
+            self.get_indicative_lmp_by_settlement_point(date="today")
+        if not end:
+            end = date + pd.DateOffset(days=1)
+
+        docs = self._get_documents(
+            report_type_id=ERCOT_INDICATIVE_LMP_BY_SETTLEMENT_POINT_RTID,
+            date=date,
+            extension="csv",
+            published_before=end,
+            published_after=date,
+            verbose=verbose,
+        )
+        df = self.read_docs(docs, parse=False, verbose=verbose)
+        return self._handle_indicative_lmp_by_settlement_point(df)
+
+    def _handle_indicative_lmp_by_settlement_point(self, df: pd.DataFrame):
+        columns_to_rename = {
+            "RepeatedHourFlag": "DSTFlag",
+            "IntervalId": "Interval Id",
+            "IntervalEnding": "Interval End",
+            "IntervalRepeatedHourFlag": "Interval Repeated Hour Flag",
+            "SettlementPoint": "Location",
+            "SettlementPointType": "Location Type",
+            "LMP": "LMP",
+        }
+        df.rename(columns=columns_to_rename, inplace=True)
+
+        def ambiguous_based_on_dstflag(df: pd.DataFrame) -> pd.Series:
+            # DSTFlag is Y during the repeated hour (after the clock has been set back)
+            # so it's False/N during DST And True/Y during Standard Time.
+            # For ambiguous, Pandas wants True for DST and False for Standard Time
+            # during repeated hours. Therefore, ambgiuous should be True when
+            # DSTFlag is False/N
+
+            # Some ERCOT datasets use a boolean, some use a string
+            if df["DSTFlag"].dtype == bool:
+                return ~df["DSTFlag"]
+            # Assume that if the DSTFlag column is a string, it's "Y" or "N"
+            else:
+                assert set(df["DSTFlag"].unique()).issubset({"Y", "N"})
+                return df["DSTFlag"] == "N"
+
+        ambiguous = ambiguous_based_on_dstflag(df)
+
+        df["RTDTimestamp"] = pd.to_datetime(df["RTDTimestamp"]).dt.tz_localize(
+            self.default_timezone,
+            ambiguous=ambiguous,
+            nonexistent="NaT",
+        )
+        df["Interval End"] = pd.to_datetime(df["Interval End"]).dt.tz_localize(
+            self.default_timezone,
+            ambiguous=ambiguous,
+            nonexistent="NaT",
+        )
+
+        df["Interval Start"] = df["Interval End"] - pd.Timedelta(minutes=5)
+        df = df.sort_values("Interval Start").reset_index(drop=True)
+        return df[
+            [
+                "Interval Start",
+                "Interval End",
+                "RTDTimestamp",
+                "Interval Id",
+                "Interval Repeated Hour Flag",
```

Comment:
im fine if we want to return this value here, but i dont think we should store `"Interval Repeated Hour Flag"` downstream in our database. we should be already encoding this data the the timezone of the the timestamp